django/db/models/functions/text.py

```diff
@@ -235,6 +236,54 @@ class RTrim(Transform):
     lookup_name = 'rtrim'
 
 
+class SHA1(MD5):
```

Comment:
I'm not sure that this inheritance chain (SHA inheriting from MD5) makes much semantic sense. Maybe there should be a mixin for the shared functionality instead?

Reply:
Agreed, an issue in creating a common mixin is that for each database it is needed for different hash functions:
- Oracle (MD5, SHA1, SHA224, SHA256, SHA384, SHA512),
- PostgreSQL (SHA1, SHA224, SHA256, SHA384, SHA512),
- MySQL (SHA224, SHA256, SHA384, SHA512).

So we can add three mixins or check functions in `as_vendor()`. :thinking:

Reply:
I've gone for three mixins for now so that we can see what it looks like...